### ISSUE TITLE
Crossref import does not distinguish print from electronic isbn/isbn

### ISSUE THREAD

**Issue by nicolasfranck** [ugent-library/biblio-backoffice#1160]
# Bug description

When importing a DOI from crossref works api,
the importer does not distinguish between electronic
and print ISBN or ISSN.

The api however has both fields `ISBN` (array of strings) and `isbn-type` (array of objects
with attributes `value` and `type`). Same goes for `ISSN` and `issn-type`.

See specification https://github.com/CrossRef/rest-api-doc/blob/master/api_format.md

# Steps to Reproduce

1. Go "Add publication"
2. Click on "Import your publication via an identifier"
3. Select "DOI" and enter `10.1007/978-3-319-20791-9_547-1`
4. Publication is added with duplicate ISBN 9783319207919. If you look at the json input record from crossref, you'll see that this is actually a print and an electronic ISBN: https://api.crossref.org/works/10.1007/978-3-319-20791-9_547-1

# Notes

The api however does not tell wether there is a correlation between the fields `isbn` and `isbn-type`.
By that I mean: is isbn-type always given if isbn is given, therefore merely acting as a more specific field
for the same data? Or does it originate from user input, and therefore it depends on the user who inserted it?
For safety maybe choose isbn-type first, and in absence try isbn?

**Comment by mietcls** [comment]
I'll ask around, but in crossref it is indeed reused... Not sure if that's correct or not.

**Comment by mietcls** [comment]
Discussed with @nicolasfranck:
Perhaps the case Nicolas gave can be an exception where the ISBN is reused as "E-ISBN", to be looked at.
According to the ISBN guidelines an ISBN must be unique for each type of carrier (Find it on https://www.isbn-international.org/content/isbn-users-manual/29 in [this document](https://www.isbn-international.org/sites/default/files/ISBN_FAQs_to_7ed_Manual_Absolutely_final.docx). So it should not be reused.

**Comment by mietcls** [comment]
@nicolasfranck can you explain what was correct / and the fix?

**Comment by nicolasfranck** [comment]
@mietcls:

* in the past we used the attribute `ISBN` from crossref to extract isbn numbers. That attribute is always an array of strings. But a previous bug, where a duplicate isbn was reported, showed that this array may contain both print and electronic isbn nummers. So without the possibility to see what is what. If a value is both print and electronic isbn, then you get duplicate values in the attribute `ISBN`. And we copied those values into our publication attribute .. `ISBN`, which is only used for print isbn's. See the problem?
* attribute `isbn-type` is, since that PR, used to extract isbn's. This is an array of object containing attributes `type` (which is either `print` or `electronic`), and `value`. This way we can now either our `ISBN` or `EISBN`.
